modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -640,15 +641,18 @@ protected ClassLoader getContextClassLoader() {
 
     ValueResolver resolver;
 
-    final Class<Object> expectedClass = getType(expectedType);
-
-    boolean isExpression = isExpression(value, parser);
-
-    resolver = isExpression
-        ? getExpressionBasedValueResolver(expectedType, (String) value, modelProperties, expectedClass)
-        : getStaticValueResolver(parameterName, expectedType, value, defaultValue, modelProperties, acceptsReferences,
-                                 expectedClass);
+    final Class<Object> expectedClass = ExtensionMetadataTypeUtils.getType(expectedType).orElse(Object.class);
 
+    if (isExpression(value, parser)) {
+      final String expression = (String) value;
+      resolver = getExpressionBasedValueResolver(expectedType, expression, modelProperties, expectedClass);
+      if (required) {
+        resolver = new RequiredExpressionParameterValueResolverWrapper(resolver, parameterName, expression);
```

Comment:
this "null-safe" wrapper should not be applied for both, the expression and static based resolvers?

Reply:
how do you do a static one which returns null?